### ISSUE TITLE
`PCAP.GATE_DURATION` produces two int32 in RAW mode and a single double in SCALED

### ISSUE THREAD

**Issue by evalott100** [PandABlocks/PandABlocks-server#47]
@dmgav found an [issue](https://github.com/PandABlocks/PandABlocks-ioc/issues/93):

With these PCAP settings
![image](https://github.com/PandABlocks/PandABlocks-server/assets/21142790/c3220a92-41ab-4f26-ad32-1e46b4c1fae4)

In `RAW` mode, `GATE_DURATION` is sent twice as two int32:
```
$ nc <ip> 8889
RAW
OK
arm_time: 2024-02-23T13:25:42.576617315Z
start_time: 2024-02-23T13:25:42.576666305Z
missed: 0
process: Raw
format: ASCII
fields:
 PCAP.GATE_DURATION int32 Value scale: 8e-09 offset: 0 units: s
 PCAP.GATE_DURATION int32 Value scale: 8e-09 offset: 0 units: s
 COUNTER1.OUT int32 Min scale: 1 offset: 0 units: 
 COUNTER1.OUT int32 Max scale: 1 offset: 0 units: 
 PCAP.TS_TRIG int64 Value scale: 8e-09 offset: 0 units: s
 COUNTER1.OUT int64 Mean scale: 1 offset: 0 units: 

 0x0032a27b 0x0032a27b 0x00026fb1 0x00026fb1 0x0032a27b 0x00000000 0x5c6bac0b 0x0000007b
 0x07735940 0x07735940 0x00026fb1 0x00026fb2 0x07a5fbbb 0x00000000 0xe466ce7f 0x00001226
 ...
```

While in `SCALED` it is sent as a single double:
```
$ nc <ip> 8889

OK
arm_time: 2024-02-23T13:25:54.448425957Z
start_time: 2024-02-23T13:25:54.448473957Z
missed: 0
process: Scaled
format: ASCII
fields:
 PCAP.GATE_DURATION double Value scale: 8e-09 offset: 0 units: s
 COUNTER1.OUT double Min scale: 1 offset: 0 units: 
 COUNTER1.OUT double Max scale: 1 offset: 0 units: 
 PCAP.TS_TRIG double Value scale: 8e-09 offset: 0 units: s
 COUNTER1.OUT double Mean scale: 1 offset: 0 units: 

 0.154739792 159677 159677 0.154739792 159677
 1 159677 159678 1.154739792 159678
 1 159678 159679 2.154739792 159679
 1 159679 159680 3.154739792 159680
 1 159680 159681 4.154739792 159681
 1 159681 159682 5.154739792 159682
 ...
```

**Comment by Araneidae** [comment]
Hmm.  For reference, the corresponding entry for `GATE_DURATION` in `config` is:
```
PCAP
    GATE_DURATION ext_out samples
```
and is processed using `CAPTURE_MODE_SCALED32`, which is pretty well the normal capture mode.

However ... I seem to remember that `ext_out samples` needs special processing, I think it needs to be used to scale other captured values, so there must be a subtle bug in how this is generated.

Capturing `sample_count` is configured here: https://github.com/PandABlocks/PandABlocks-server/blob/e5a5dab5512a008d6310e63bc2dbaad4a845c736/server/prepare.c#L461-L463 it looks like the decision to send the field separately is here: https://github.com/PandABlocks/PandABlocks-server/blob/e5a5dab5512a008d6310e63bc2dbaad4a845c736/server/prepare.c#L392-L395.

My best guess is that `sample_count_is_anonymous()` is returning true which means that `ensure_sample_count()` defined here: https://github.com/PandABlocks/PandABlocks-server/blob/e5a5dab5512a008d6310e63bc2dbaad4a845c736/server/capture.c#L390-L391 is failing to spot that is already being captured.  I'll try and reproduce unless you can spot the obvious mistake!

**Comment by coretl** [comment]
Might have something to do with https://github.com/PandABlocks/PandABlocks-server/blob/e5a5dab5512a008d6310e63bc2dbaad4a845c736/server/ext_out.c#L314

Did we change the ext_out type of the field from samples to something else when we changed its name?

**Comment by EmilioPeJu** [comment]
I'm not able to reproduce this problem in my branch, either there is some extra configuration needed, or I accidentally fixed this

**Comment by Araneidae** [comment]
Turns out we _did_ change the type of `EXT_OUT_SAMPLES` here: 2b39c87.  Fixed by @EmilioPeJu in #52.
